### PR TITLE
geo/geomfn: implement ST_AsHexEWKB({geometry,text}), ST_AsHexEWKB({ge…

### LINKED ISSUE #48880: geo/geomfn: implement ST_AsHexEWKB({geometry,text})
Implement `ST_AsHexEWKB` on arguments {geometry,text}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_AsHexEWKB.html).

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end.
* Ensure the documentation is regenerated by calling `make buildshort`.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-05-20T02:07:31Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by hueypark**
…ography,text})

Fixes #48880

Release note (sql change): Implemented the geometry based builtins `ST_AsHexEWKB` with encoding.

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/geospatial [hunk 1]
```diff
@@ -249,25 +251,27 @@ SELECT
   ST_AsEWKB(geog),
   ST_AsHexWKB(geog),
   ST_AsHexEWKB(geog),
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/geospatial:253]: this output is getting a little harder to read. do you mind moving HexWKB/HexEWKB to a new query?

File: pkg/sql/sem/builtins/geo_builtins.go [hunk 2]
```diff
@@ -695,6 +696,72 @@ var geoBuiltins = map[string]builtinDefinition{
...
+				b, err := ewkb.Marshal(geomT, byteOrder)
+				if err != nil {
+					return nil, err
+				}
+
+				return tree.NewDString(fmt.Sprintf("%X", b)), nil
+			},
+			Info: infoBuilder{
+				info: "Returns the EWKB representation in hex of a given Geometry. " +
+					"This variant has a second argument denoting the encoding.",
```
Comment (otan) [inline pkg/sql/sem/builtins/geo_builtins.go:728]: Do you mind adding what `xdr` and `ndr` means? The other builtin has this annotation.

**Comment by hueypark** [comment]
Thank you for the review!

File: pkg/sql/sem/builtins/geo_builtins.go (hunk 2 above)
Comment (hueypark) [reply pkg/sql/sem/builtins/geo_builtins.go]: Done.

File: pkg/sql/logictest/testdata/logic_test/geospatial (hunk 1 above)
Comment (hueypark) [reply pkg/sql/logictest/testdata/logic_test/geospatial]: Done.

**Review by otan — APPROVED** [review]

**Pull request merged**